blog/serialize-a-style-using-your-chakra-ui-theme.md

```diff
@@ -0,0 +1,113 @@
+# Serialize a style using your Chakra UI theme
+
+A time ago I had to build a _subscription form_ using the [Spreedly iFrame API](https://docs.spreedly.com/reference/iframe/v1/) to allow a user to use its credit card in a safe way.
+
+After a while, I found myself into an issue with the card number and the CVV number inputs because Spreedly inserts an `iframe` and the inputs are not under my control (and my CSS).
+
+## The easy solution
+
+Afortunately Spreedly accepts a style inline string via the [`setStyle`](https://docs.spreedly.com/reference/iframe/v1/#setstyle) function.
+
+```javascript
+Spreedly.on("ready", () => {
+  Spreedly.setStyle("number", "width:225px;  height:35px;");
+});
+```
+
+A little ugly but a solution at least. Lets see how to improve this...
+
+## Using the theme with strings
+
+We can use the [`useTheme` hook](https://chakra-ui.com/docs/hooks/use-theme) to get all the tokens we had defined in the theme.
+
+```javascript
+import {useEffect} from 'react'
+import {useTheme} from "@chakra-ui/react"
+
+const buildStyles = theme => `
+  border: 2px solid ${theme.colors.gray[300]},
+  color: ${theme.colors.black},
+  lineHeight: ${theme.sizes[5]}
+`
+
+const MyComp = () => {
+  const theme = useTheme();
+  useEffect(() => {
+    Spreedly.on("ready", () => {
+      Spreedly.setStyle("number", buildStyles(theme));
+    });
+  }, []);
+
+  return <>...</>;
+};
+```
+
+This is similar to `styled-components` using the string templates to create styles.
+
+💡 Tip: remember that your component should be wrapped by a `ChakraProvider` to get the `theme` object.
+
+## Using a CSS object
+
+I looked for a nicer way to handle a CSS object in JavaScript instead of a one big string. Chakra UI uses [emotion](https://emotion.sh/) under the hook to build the CSS classes, so I found this solution:
```

Comment:
```suggestion
I looked for a nicer way to handle a CSS object in JavaScript instead of using one big string. Chakra UI uses [emotion](https://emotion.sh/) under the hook to build the CSS classes, so I found this solution:
```